### PR TITLE
Gradient accumulation 

### PR DESCRIPTION
**Pull request by amogh-gulati**
Added gradient accumulation support with configurable gradient_accumulation_steps and effective batch size logging.
based on this [writeup](https://www.kozodoi.me/blog/gradient-accumulation-in-pytorch)

### CODE REVIEW

**Comment by jder** [comment]
@codex review

File: src/ocean_emulators/train.py [hunk 1]
```diff
@@ -314,6 +314,22 @@ def __init__(self, cfg: TrainConfig) -> None:
             cfg.resume_ckpt_path, cfg, finetune=cfg.finetune
         )
 
+        # Log effective batch size
+        effective_batch_size = cfg.batch_size * cfg.gradient_accumulation_steps
+        logger.info(
+            f"Effective batch size: {effective_batch_size} "
+            f"(batch_size={cfg.batch_size} × "
+            f"gradient_accumulation_steps={cfg.gradient_accumulation_steps} × "
```
Comment (jder) [inline src/ocean_emulators/train.py:271]:
```suggestion
            f"gradient_accumulation_steps={cfg.gradient_accumulation_steps})"
```

File: src/ocean_emulators/train.py [hunk 2]
```diff
@@ -314,6 +314,22 @@ def __init__(self, cfg: TrainConfig) -> None:
...
+        )
+        if self.is_wandb_enabled():
+            self.wandb_logger.log(
+                {
+                    "config/effective_batch_size": effective_batch_size,
+                    "config/gradient_accumulation_steps": cfg.gradient_accumulation_steps,
+                },
+                step=0,
+            )
```
Comment (jder) [inline src/ocean_emulators/train.py:279]: We already log the full config in wandb_logger.setup_run -- do we also need to log the accum steps here? Logging the effective batch size makes sense to me!

**Review by jder — APPROVED** [review]
Thanks! Just a couple tiny suggestions. I also agree with Codex's suggestion.
